lib/fileio/translators/translatorMesh.cpp

```diff
@@ -171,6 +171,19 @@ TranslatorMeshRead::TranslatorMeshRead(const UsdGeomMesh& mesh,
         return;
     }
 
+    // set mesh name
+    const auto& primName = prim.GetName().GetString();
+    const auto shapeName = TfStringPrintf("%sShape", primName.c_str());
+    meshFn.setName(MString(shapeName.c_str()), false, &stat);
+
+    if (!stat) {
+          *status = stat;
+          return;
+    }
+
+	// store the path
+    m_shapePath = prim.GetPath().AppendChild(TfToken(shapeName));
```

Comment:
After the mesh function set is created, we set it's name and then store the SdfPath that is created off of it.